src/expandableCalendar/WeekCalendar/index.tsx

```diff
@@ -149,74 +85,77 @@ class WeekCalendar extends Component<WeekCalendarProps, State> {
+  const keyExtractor = useCallback((_, index: number) => index.toString(), []);
 
-  renderWeekDaysNames = () => {
+  const renderWeekDaysNames = useMemo(() => {
     return (
-      <WeekDaysNames 
-        firstDay={this.props.firstDay} 
-        style={this.style.dayHeader} 
+      <WeekDaysNames
+        firstDay={firstDay}
+        style={dayHeader}
       />
     );
-  };
+  },[firstDay, dayHeader]);
 
-  render() {
-    const {allowShadow, firstDay, hideDayNames, current, context} = this.props;
-    const {items} = this.state;
-    const extraData = Map({
-      current,
-      date: context.date,
-      firstDay
-    });
+  const extraData = Map({
+    current,
+    date: context.date,
+    firstDay
+  });
 
-    return (
-      <View
-        testID={this.props.testID}
-        style={[allowShadow && this.style.containerShadow, !hideDayNames && this.style.containerWrapper]}
-      >
-        {!hideDayNames && (
-          <View style={[this.style.week, this.style.weekCalendar]}>
-            {/* {this.props.weekNumbers && <Text allowFontScaling={false} style={this.style.dayHeader}></Text>} */}
-            {this.renderWeekDaysNames()}
-          </View>
-        )}
-        <FlatList
-          ref={this.presenter.list}
-          data={items}
-          extraData={extraData}
-          style={this.style.container}
-          horizontal
-          showsHorizontalScrollIndicator={false}
-          pagingEnabled
-          scrollEnabled
-          renderItem={this.renderItem}
-          keyExtractor={this.keyExtractor}
-          initialScrollIndex={NUMBER_OF_PAGES}
-          getItemLayout={this.getItemLayout}
-          onScroll={this.onScroll}
-          onMomentumScrollEnd={this.onMomentumScrollEnd}
-        />
-      </View>
-    );
+  return (
+    <View
+      testID={props.testID}
+      style={[allowShadow && defaultStyle.containerShadow, !hideDayNames && defaultStyle.containerWrapper]}
+    >
+      {!hideDayNames && (
+        <View style={[defaultStyle.week, defaultStyle.weekCalendar]}>
+          {renderWeekDaysNames}
+        </View>
+      )}
+          <FlashList
+            ref={presenter.list as unknown as LegacyRef<FlashList<string>>}
+            data={items}
+            extraData={extraData}
+            contentContainerStyle={defaultStyle.container}
+            horizontal
+            showsHorizontalScrollIndicator={false}
+            pagingEnabled
+            scrollEnabled
+            renderItem={renderItem}
+            keyExtractor={keyExtractor}
+            initialScrollIndex={NUMBER_OF_PAGES}
+            onScroll={onScroll}
+            onMomentumScrollEnd={onMomentumScrollEnd}
+            estimatedItemSize={30}
+          />
+    </View>
+  );
+};
+
+WeekCalendar.displayName = 'WeekCalendar';
+WeekCalendar.defaultProps = {
+  firstDay: 0,
+  allowShadow: true
+};
+
+function shouldUpdate(prevProps: WeekCalendarProps, nextProps: WeekCalendarProps) {
```

Comment:
If we migrated to functional component, why not do this in `useEffect`?

Reply:
because i wont have the prevProps there, and i wouldn't know what the previous date was

Reply:
If the useEffect with `context` dependency is called, you know the context change. Second, the previous date you can keep in `useRef` and check `sameWeek` against it - update it only after.